mne/realtime/mock_lsl_stream.py

```diff
@@ -0,0 +1,98 @@
+# Authors: Teon Brooks <teon.brooks@gmail.com>
+#
+# License: BSD (3-clause)
+import time
+from multiprocessing import Process
+
+from numpy.random import rand
+import numpy as np
+
+from ..utils import _check_pylsl_installed
+
+
+class MockLSLStream:
+    """Mock LSL Stream.
+
+    Parameters
+    ----------
+    host : str
+        The LSL identifier of the server.
+    n_channels : int
+        Number of channels.
+    ch_type : str
+        The type of device that is being mocked.
+    sfreq : float
+        Sampling frequency of mock device.
+    testing : bool
+        Setting used to determine whether the data stream will be used
+        for testing where the data is uniform and expected or with some
+        random variation. Default is False.
+    """
+
+    def __init__(self, host, n_channels=8, ch_type="eeg", sfreq=100,
+                 testing=False):
```

Comment:
I would instead suggest that it take a `raw` object as an argument. Then you don't need n_channels, ch_type, sfreq etc.

Reply:
agreed, it should take any `raw` instance (subject to any necessary constraints of the streamer, e.g. only EEG channels or something) and "play it back"

Reply:
that is totally possible. looking at the code, i realize that the name mockrtclient is either a misnomer or it is working as both the server and the client. I feel that these two functionality should be separated. what i mean is that the client shouldn't be serving up data, it should be a listener that can pass the data along to other functionalities.

there should be stream generator that takes a raw file and send it to LSL or TCP/IP. The client should then be able to listen using their protocols. it could combine with the existing `stim_server_client` module for the latter protocol

Reply:
I agree it would be cleaner to have a mock server emulate mne_rt_server but this is written in CPP. Writing a mock server to emulate this may not be so straightforward, and certainly was not for someone who had learned python 10 days ago ;-) But this could perhaps be a point of collaboration with @LorenzE since he may be familiar with the MNE-CPP code and the protocol for mne_rt_server.